### PR TITLE
[4229575092] Added Personal Avatars page

### PR DESCRIPTION
**Pull request by gadamyan**
## Description

-   Added Personal Avatars page
-   Added PersonalAvatarDownloader class for fetching the avatars
-   Changed the navigation logic to support the personal avatars page
-   Blueprint refactoring

#### Added

-   Added Personal Avatars page
-   Added PersonalAvatarDownloader class for fetching the avatars

#### Updated

-   Changed the navigation logic to support the personal avatars page

## How to Test

-   After logging in, the player should be redirected to the Persona Avatars page

## Checklist

-   [ ] Tests written or updated for the changes.
-   [ ] Documentation is updated.
-   [ ] Changelog is updated.
-   [ ] QA Testing is updated.

### CODE REVIEW

File: Source/RpmAvatarCreator/Private/Requests/Endpoints.cpp [hunk 1]
```diff
@@ -42,6 +44,16 @@ FString FEndpoints::GetAvatarPublicUrl(const FString& AvatarId)
 	return FString::Format(TEXT("{0}/{1}.glb"), {AVATAR_API_V2_ENDPOINT, AvatarId});
 }
 
+FString FEndpoints::GetRenderEndpoint(const FString& AvatarId)
+{
+	return FString::Format(TEXT("{0}/{1}.png"), {RENDER_URL_PREFIX, AvatarId});
+}
+
+FString FEndpoints::GetPersonalAvatarsEndpoint(const FString& UserId)
```
Comment (HarrisonHough) [inline Source/RpmAvatarCreator/Private/Requests/Endpoints.cpp:52]: This function gets the URL you use to retrieve the avatars associated to a User account right? 
If so perhaps a better name is GetUserAvatarsEndpoint 
To me Personal Avatars is a bit misleading

File: Source/RpmAvatarCreator/Private/Extractors/PersonalAvatarExtractor.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		if (!FJsonSerializer::Deserialize(Reader, JsonObject) || !JsonObject->HasField(JSON_FIELD_DATA))
+		{
+			return {};
+		}
+	
+		return JsonObject->GetArrayField(JSON_FIELD_DATA);
+	}
+}
+
+TArray<FRpmPersonalAvatar> FPersonalAvatarExtractor::ExtractPersonalAvatars(const FString& JsonString)
```
Comment (HarrisonHough) [inline Source/RpmAvatarCreator/Private/Extractors/PersonalAvatarExtractor.cpp:30]: As mentioned in my other comment I think UserAvatarExtractor would be a better name for this. 
In general we want to provide personal avatars at RPM but in this case we specifically want to get avatars associated with a User

File: Source/RpmAvatarCreator/Private/RpmDefaultAvatarDownloader.cpp [hunk 3]
```diff
@@ -56,7 +52,7 @@ TArray<FRpmDefaultAvatarData> URpmDefaultAvatarDownloader::GetFilteredAvatars()
 	TArray<FRpmDefaultAvatarData> DefaultAvatars;
 	for (const auto& Id : TemplateAvatarIds)
 	{
-		if (TemplateProperties.Contains(Id) && ImageMap.Contains(Id) && IsAvatarFiltered(TemplateProperties[Id], SelectedBodyType, SelectedGender))
+		if (TemplateProperties.Contains(Id) && ImageMap.Contains(Id) && IsAvatarFiltered(TemplateProperties[Id], SelectedGender))
```
Comment (HarrisonHough) [inline Source/RpmAvatarCreator/Private/RpmDefaultAvatarDownloader.cpp:55]: minor thing but you could potentially make this if statement more readable by putting this condition into some function with a description name. 
EG something like 
bool IsValidAvatar(string id, EAvatarGender selectedGender) 
{
return TemplateProperties.Contains(Id) && ImageMap.Contains(Id) && IsAvatarFiltered(TemplateProperties[Id], SelectedGender);
}

**Review by HarrisonHough — APPROVED** [review]
Code looks good. Cant test in UE atm unfortunately. 
Some minor suggestions in the comments.

**Comment by srcnalt** [comment]
After starting to edit an avatar if I clock on back and create a new avatar it takes me back to edit screen.

**Comment by gadamyan** [comment]
> After starting to edit an avatar if I clock on back and create a new avatar it takes me back to edit screen.

Will check tomorrow

**Pull request merged**